query-algebrizer/src/lib.rs

```diff
@@ -75,15 +75,7 @@ pub fn algebrize(schema: &Schema, parsed: FindQuery) -> Result<AlgebraicQuery> {
     let mut cc = cc::ConjoiningClauses::default();
     let where_clauses = parsed.where_clauses;
     for where_clause in where_clauses {
-        match where_clause {
-            WhereClause::Pattern(p) => {
-                cc.apply_pattern(schema, p);
-            },
-            WhereClause::Pred(p) => {
-                cc.apply_predicate(schema, p)?;
-            },
-            _ => unimplemented!(),
-        }
+        cc.apply_clause(schema, where_clause);
```

Comment:
This ignores the result, doesn't it?  You want the trailing `?` or some other chaining operator.